controller/linodemachine_controller.go

```diff
@@ -295,6 +317,11 @@ func (r *LinodeMachineReconciler) reconcilePreflightCreate(ctx context.Context,
 		}
 		return retryIfTransient(err)
 	}
+
+	if linodeInstance == nil {
```

Comment:
Is this check necessary? Based on the above implementation, linodeInstance will be nil only when the err is not nil and we are already performing that check above

Reply:
updated with newer implementation